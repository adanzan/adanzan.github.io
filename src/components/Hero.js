import * as React from "react";
import { useState } from "react";

import styles from "../styles/Hero.module.css";
import portrait from "../resources/images/portrait.png";

export default function Hero () {
  //On-hover, On-leave, change the visibility of name 
    // 1 = Anthony
    // 2 = Danzan
    // 0 = None Selected
  const [mouseHover, setMouseHover] = useState(0);

  const HOVER_OPACITY = 0.1;

  return (
    <div className={styles.Hero}>
      <div className={styles.textSection}>
        <h1>Hi, I'm</h1>
        {/* Reduces opacity of firstName that is not selected */}
        <div className={styles.firstName} onMouseLeave={() => {setMouseHover(0)}}>
          <h1 className={styles.firstNameLeft} style={{opacity: mouseHover === 2? HOVER_OPACITY : 1}}
              onMouseEnter={()=> {setMouseHover(1)}}>Anthony</h1>
          <h1 className={styles.firstNameRight} style={{opacity: mouseHover === 1? HOVER_OPACITY : 1}}
              onMouseEnter={()=> {setMouseHover(2)}}>Danzan</h1>
        </div>
        <h1>Achit-Erdene</h1>
      </div>
      <img className={styles.image} src={portrait} alt="Me"/>
    </div>
  );
}
