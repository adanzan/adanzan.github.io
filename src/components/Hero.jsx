import * as React from "react";

import styles from "../styles/Hero.module.css";
import portrait from "../resources/images/me.jpeg";

export default function Hero() {
	return (
		<div className={styles.Hero}>
			<div className={styles.textSection}>
				<h1>Hello, I&apos;m</h1>
				<h1>Anthony</h1>
				<h1>Danzan</h1>
				<h1>Achit-Erdene</h1>
				<p className={styles.brief}>
					I'm a Software Developer based in Boston, MA.
				</p>
			</div>
			<img className={styles.image} src={portrait} alt="Me" />
		</div>
	);
}
