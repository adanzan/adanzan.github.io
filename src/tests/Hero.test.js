/* eslint no-unused-vars: 0 */
import * as React from "react";
import { render } from "@testing-library/react";

import Hero from "../components/Hero"; 

let testRender;

describe("Hero component", () => {
	beforeEach(() => {
		testRender = render(<Hero />);
	});
	test("matches snapshot", () => {
		expect(testRender).toMatchSnapshot();
	});
	describe("displays", () => {
		test("first name", () => {
			expect(testRender.getByText("Anthony"));
		});
		test("image", () => {
			expect(testRender.getByRole("img", "Me"));
		});
		test("brief", () => {
			expect(testRender.getByText(/Software Developer/));
		});
	});
});