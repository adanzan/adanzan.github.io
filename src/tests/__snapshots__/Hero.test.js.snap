// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hero"
      >
        <div
          class="textSection"
        >
          <h1>
            Hello, I'm
          </h1>
          <div
            class="firstName"
          >
            <h1
              class="firstNameLeft"
              style="opacity: 1;"
            >
              Anthony
            </h1>
            <h1
              class="firstNameRight"
              style="opacity: 1;"
            >
              Danzan
            </h1>
          </div>
          <h1>
            Achit-Erdene
          </h1>
          <p
            class="brief"
          >
            I am a Software Developer based in Massachusetts.
          </p>
        </div>
        <img
          alt="Me"
          class="image"
          src="portrait.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hero"
    >
      <div
        class="textSection"
      >
        <h1>
          Hello, I'm
        </h1>
        <div
          class="firstName"
        >
          <h1
            class="firstNameLeft"
            style="opacity: 1;"
          >
            Anthony
          </h1>
          <h1
            class="firstNameRight"
            style="opacity: 1;"
          >
            Danzan
          </h1>
        </div>
        <h1>
          Achit-Erdene
        </h1>
        <p
          class="brief"
        >
          I am a Software Developer based in Massachusetts.
        </p>
      </div>
      <img
        alt="Me"
        class="image"
        src="portrait.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
